packages/devtools_app/lib/src/screens/debugger/program_explorer_controller.dart

```diff
@@ -20,7 +20,9 @@ class ProgramExplorerController extends DisposableController
   /// outline view.
   ProgramExplorerController({
     this.showCodeNodes = false,
-  });
+    ListValueNotifier<VMServiceObjectNode>? rootObjectNodesOverride,
+  }) : _rootObjectNodes = rootObjectNodesOverride ??
+            ListValueNotifier<VMServiceObjectNode>([]);
```

Comment:
Is there a better way we can do this that isn't changing the constructor for the sake of a test? For example, maybe creating a visibleForTesting getter for _rootObjectNodes that can be overridden in a TestProgramExplorerController class?

Reply:
You can't override a private property/method if you're not in the same library, so we'd have to make `_rootObjectNodes` public at the very least.

Reply:
There is already a public getter `rootObjectNodes`. This could be overriden on the TestProgramExplorer class.

Reply:
Okay nevermind. I see the issue. I think making rootObjectNodesInternal public and visible for testing is a better solution than changing the constructor. lgtm.